### PR TITLE
Docs update

### PR DESCRIPTION
**Pull request by maririos**
Update documentation to make it consistent with our official documentation and the Release Planner. Came to my radar with PR https://github.com/Azure/azure-sdk-for-net/pull/45892.

### CODE REVIEW

**Comment by maririos** [comment]
FYI @abatishchev

File: doc/DataPlaneCodeGeneration/AzureSDKGeneration_Prerequistites.md [hunk 1]
```diff
@@ -12,10 +12,8 @@
 
 ## Set up TypeSpec project
 
-Make sure you have typespec project in API specification repo, e.g. [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs) repo or you can follow [TypeSpec Getting Started](https://github.com/microsoft/typespec/#using-node--npm) to initialize your typespec project, and refer to [TypeSpec Structure Guidelines](https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/typespec-structure-guidelines.md) to configure your typespec project.
+Make sure you have a typespec project in the `main` branch of the public API specification repository [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). You can find more information about creating an API spec on TypeSpec [here](https://aka.ms/azsdk/typespec).
```
Comment (jsquire) [inline doc/DataPlaneCodeGeneration/AzureSDKGeneration_Prerequistites.md:15]:
```suggestion
Make sure you have a TypeSpec project in the `main` branch of the public API specification repository [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). You can find more information about creating an API spec on TypeSpec [here](https://aka.ms/azsdk/typespec).
```

nit:  branding

File: doc/DataPlaneCodeGeneration/Create_Samples_Guidance.md [hunk 2]
```diff
@@ -48,6 +54,10 @@ public partial class Sample1_HelloWorld
 }
 ```
 
+## Snippets
+
+Snippets are the great way to reuse the sample code. Snippets allow us to verify that the code in our samples and READMEs is always up to date, and passes unit tests. We have added the snippet [here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/template/Azure.Template/tests/Samples/Sample1_HelloWorld.cs#L21) in a sample and used it in the [README](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/template/Azure.Template/README.md#get-secret). Please refer to [Updating Sample Snippets](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets) to add snippets in your samples.
```
Comment (jsquire) [inline doc/DataPlaneCodeGeneration/Create_Samples_Guidance.md:59]:
```suggestion
All samples in the Azure SDK for .NET repository are required to use snippets when demonstrating code.  Snippets allow us to verify that the code in our samples and READMEs is always up to date and passes unit tests. We have added the snippet [here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/template/Azure.Template/tests/Samples/Sample1_HelloWorld.cs#L21) in a sample and used it in the [README](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/template/Azure.Template/README.md#get-secret). Please refer to [Updating Sample Snippets](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets) to add snippets in your samples.
```

**Review by jsquire — APPROVED** [review]
Thank you for doing this, @maririos!

**Pull request merged**